packages/radix-ui-themes/src/helpers/first-child-might-adopt-subtree.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+/**
+ * This is a helper function that is used when a component supports `asChild`
+ * using the `Slot` component but its implementation contains nested DOM elements.
+ *
+ * Using it ensures if a consumer uses the `asChild` prop, the elements are in
+ * correct order in the DOM, adopting the intended consumer `children`.
+ */
+export function firstChildMightAdoptSubtree(
+  options: { asChild: boolean | undefined; children: React.ReactNode },
+  content: React.ReactNode | ((children: React.ReactNode) => React.ReactNode)
```

Comment:
Added the simple case where we don't need children as convenience (for cases like `Avatar` where the component actually doesn't take `children` (other than for `asChild` purposes).